src/hooks/api/useAuthentication.tsx

```diff
@@ -0,0 +1,41 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useMemo } from "react"
+import { useSignInWithEthereum } from "../useSignInWithEthereum"
+import { AuditorProfile } from "./auditors/index"
+import { useProfile } from "./auditors"
+import { useSignOut } from "./auditors/useSignOut"
+
+type AuthenticationContextType = {
+  isLoading: boolean
+  isError: boolean
+  isAuthenticated: boolean
+  signature?: string
+  authenticate: () => Promise<void>
+  signOut: () => void
+  profile?: AuditorProfile
+  status: "error" | "idle" | "loading" | "success"
+}
+
+const AuthenticationContext = createContext<AuthenticationContextType>({} as AuthenticationContextType)
+
+export const useAuthentication = () => useContext(AuthenticationContext)
+
+export const AuthenticationContextProvider: React.FC<PropsWithChildren<unknown>> = ({ children }) => {
+  const { signIn, isLoading, signature } = useSignInWithEthereum()
```

Comment:
`signature` should be exported? I see it part of `AuthenticationContextType`

Reply:
not needed anymore, I'll remove it from the context type